### PR TITLE
Return the challenges of a student

### PR DESCRIPTION
**Pull request by thekosiguy**
.get('/:id') route also returns the challenges corresponding to a specific student.

### CODE REVIEW

File: routes/students.js [hunk 1]
```diff
@@ -14,7 +15,16 @@ router.get('/:id', async function (req, res) {
     }
   })
 
-  res.json({student: student})
+  const studentChallenges = await ModuleChallenge.findAll({
```
[the remark this answers is not in the document]
Comment (thekosiguy) [reply routes/students.js]: I've made the required changes, please review this pull request again.

File: cypress/integration/studentInfo.spec.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            expect(res.body.student.ModuleChallenges[1].dueDate).to.contain('2021-01-08')
+            expect(res.body.student.ModuleChallenges[1].submissionDate).to.contain('2021-01-08')
+            expect(res.body.student.ModuleChallenges[1].createdAt).to.contain('2021-01-08')
+            expect(res.body.student.ModuleChallenges[1].updatedAt).to.contain('2021-01-08')
+            
+            expect(res.status).eq(200)
+        })
+    })
+
+    it('Retrieves data for a student with an id of 2', function(){
```
Comment (dearshrewdwit) [inline cypress/integration/studentInfo.spec.js:37]: There is no benefit to having additional tests asserting on a successful request to the student endpoint.

What about if the student does not exist? What does the response look like?

File: package.json [hunk 3]
```diff
@@ -23,7 +23,7 @@
     "sequelize-cli": "^6.2.0"
   },
   "devDependencies": {
-    "cypress": "^8.3.0",
+    "cypress": "^8.3.1",
```
Comment (dearshrewdwit) [inline package.json:26]: Please undo/remove from this PR - all dependency version upgrades should be on a separate PR

File: cypress/integration/studentInfo.spec.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        cy.task('taskCreateModuleChallenges')
+        })
+    })
+
+    it('Retrieves data for a student with an id of 1', function(){
+        cy.request({
+            method: 'GET',
+            url: '/api/students/1'
+        }).should((res) => {
+            expect(res.body.student.ModuleChallenges[0].challengeName).to.eq('bank')
```
Comment (dearshrewdwit) [inline cypress/integration/studentInfo.spec.js:15]: simplify:

```js
const bankChallenge = res.body.student.ModuleChallenges[0]

expect(bankChallenge.challengeName).to.eq('bank')
expect(bankChallenge.language).to.eq('node')
// etc
```

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]
Good. See comments

File: cypress/integration/studentInfo.spec.js (hunk 2 above)
Comment (thekosiguy) [reply cypress/integration/studentInfo.spec.js:37]: fair enough, will work on that

**Comment by thekosiguy** [comment]
Please review this PR again.

**Comment by thekosiguy** [comment]
Doesn't null represent an empty object?

File: cypress/integration/studentInfo.spec.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      expect(res.status).eq(200);
+    });
+  });
+
+  it("Student does not exist", function(){
+    cy.request({
+      method: "GET",
+      url: "/api/students/9"
+    }).should((res) => {
+      expect(res.body.student).to.eq(null);
```
Comment (dearshrewdwit) [inline cypress/integration/studentInfo.spec.js:45]: Change `null` to `{}`

File: routes/students.js [hunk 6]
```diff
@@ -4,17 +4,18 @@ const { Student } = require("../models");
...
 router.get("/:id", async function (req, res) {
-  const student = await Student.findOne({
+  const student = await Student.findOne({include: {all: true },
     where: {
       id: req.params.id
     }
   });
 
-  res.json({student: student});
+  res.json({ student: student });
```
Comment (dearshrewdwit) [inline routes/students.js:18]: Avoid relying on `null` values.
```js
const student = student || {}
res.json({ student: student });
```

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]

File: cypress/integration/studentInfo.spec.js (hunk 5 above)
Comment (thekosiguy) [reply cypress/integration/studentInfo.spec.js:45]: did so yesterday lol just haven't pushed yet, will do that now

**Review by dearshrewdwit — APPROVED** [review]

**Review by RyanVanDijck — APPROVED** [review]

**Pull request merged**